### ISSUE TITLE
`Shell` - Add accessibility tests and resolve violations

### ISSUE THREAD

**Issue by SimonFinney** [carbon-design-system/ibm-security#434]
## Description

Add accessibility tests and resolve violations. (based on the original unedited version of this review: https://github.com/carbon-design-system/ibm-security/pull/285#pullrequestreview-370603953)

### Skip to content link

On the `a` element:
- [ ] **DAP Violation: Each form control must have an associated label.** (Form control element `a` is missing associated label.)  [Checkpoint 1.3.1 - Info and Relationships](http://www.ibm.com/able/guidelines/ci162/info_and_relationships.html); [Checkpoint 4.1.2 Name, Role, Value ](http://www.ibm.com/able/guidelines/ci162/name_role_value.html)
- [ ] **DAP Violation: All content must reside within a HTML5 Sectioning element, WAI-ARIA landmark or labelled region role.** (Ophan content is not in a WAI-ARIA landmark for labelled region role.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

Styles:
- [ ] The link is not properly hidden when not-focused -- it still has a z-index, height/width, and visibility just not content due to the `clip` style. And `clip: auto` is the only style being used to "show" the button when focused. (See this comment for more context: https://github.com/carbon-design-system/ibm-security/pull/398#issuecomment-599610372)

### "Return to..." banner

On the `a` element:
- [ ] **DAP Violation: All content must reside within a HTML5 Sectioning element, WAI-ARIA landmark or labelled region role.** (Ophan content is not in a WAI-ARIA landmark for labelled region role.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

### Toolbar

On the `a#security--nav__list__item` element:
- [ ] **DAP Violation: Element ids must be unique within the document.** (The ids `security--nav__list__item` are used repeatedly, and they should not be.) [Checkpoint 4.1.1 - Parsing](http://www.ibm.com/able/guidelines/ci162/parsing.html) **NOTE:** Was starting to address this in #385 , so may be more appropriate to defer to that PR & fixing the inner `NavListItem` etc. components)

On the `Toolbar`'s `aside` element:
- [ ] **DAP Violation: The WAI-ARIA role(s) and/or attribute(s) must be valid for the relevant element.** (The `role="menu"` is not valid for an `aside`.) [Checkpoint 4.1.2 - Name, Role, Value ](http://www.ibm.com/able/guidelines/ci162/name_role_value.html)

On the `Toolbar`'s `nav.security--toolbar` wrapping element:
- [ ] **DAP Violation: Multiple 'navigation' sections or landmarks must have unique labels specified with 'aria-label' or 'aria-labelledby'.** (The `nav` element needs a unique label.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

### Notifications

For individual notifications:
- [ ] **DAP Violation: Each form control must have associated label.** (The close buttons for each notification, `button.security--header__notification__button`, are missing labels.)  [Checkpoint 1.3.1 - Info and Relationships](http://www.ibm.com/able/guidelines/ci162/info_and_relationships.html); [Checkpoint 4.1.2 Name, Role, Value ](http://www.ibm.com/able/guidelines/ci162/name_role_value.html)
- [ ] **Interaction:** The "close" notification button has no *visible* hover/focus state.
- [ ] **Interaction:** The "close" notification button does not do anything when clicked.

For the popover footer:
- [ ] **DAP Violation: A FOOTER element or element with WAI-ARIA 'contentinfo' role is not permitted without the presence of a 'main' role.** (There is a `footer` element and no `main` in the popover.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

For the wrapper `div.security--shell__unstable__popover`:
- [ ] **DAP Violation: ASIDE elements or elements having a WAI-ARIA 'complementary' role must have a label specified with 'aria-label' or 'aria-labelledby'.** (The element has `role="complementary"`, which makes it like an `aside` element, but it is missing a label that ties it to the content it is complementing.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

### Profile popover

For the profile image:
- [ ] **DAP Violation: Contrast of foreground and background must be sufficient to meet WCAG AA requirements.** (The initials text on the gray background, serving as a default, only has **a contrast of 2.10** & therefore does not meet AA contrast requirements.) [Checkpoint 1.4.3 - Contrast Minimum](http://www.ibm.com/able/guidelines/ci162/contrast.html)

For every other line of text the profile popover:
- [ ] **DAP Violation: Contrast of foreground and background must be sufficient to meet WCAG AA requirements.**  (The text in this popover only has a **contrast of 2.62**, which is too low for 12px text.) [Checkpoint 1.4.3 - Contrast Minimum](http://www.ibm.com/able/guidelines/ci162/contrast.html)

For the wrapper `div.security--shell__unstable__popover`:
- [ ] **DAP Violation: ASIDE elements or elements having a WAI-ARIA 'complementary' role must have a label specified with 'aria-label' or 'aria-labelledby'.** (The element has `role="complementary"`, which makes it like an `aside` element, but it is missing a label that ties it to the content it is complementing.) [Checkpoint 2.4.1 - Bypass Blocks](https://www.ibm.com/able/guidelines/ci162/bypass_blocks.html)

## Acceptance criteria

- [ ] Accessibility tests are added to the `Shell`
- [ ] Violations are resolved

**Comment by Deriohig** [comment]
Can a list of the violations be added to this story to help reviewers understand what is needed to push this story through ? Is there a story around unit testing that should be linked to this issue too?

**Comment by jendowns** [comment]
@Deriohig the bulk comes from this review: https://github.com/carbon-design-system/ibm-security/pull/285#pullrequestreview-370603953 I think @SimonFinney forgot to link to it above. I will post an un-edited version of the review in the issue description in a moment. 👍

**Comment by joemc11** [comment]
Yes this should be reopened, still a valid defect
